### ISSUE TITLE
out-of-bandwidth notification

### ISSUE THREAD

**Issue by roadscape** [steemit/condenser#2443]
when a user is out of bandwidth, condenser should display a message notifying them that their transactions will not go through until (a) their bandwidth recharges, (b) chain usage is not at "peak load", or (c) they power up. there should be a link to the FAQ with a more detailed explanation.

a good place for this might be a horizontal bar below the header. not tall but impossible to miss.

**Comment by sneak** [comment]
After the bitcoin mess, saying anything about “peak load” when blocks aren’t even 25% full is both untrue and a PR disaster.

I am not opposed to doing this, but I think it becomes very low priority once the reserve ratio algorithm is fixed, which must happen (and soon).

**Comment by roadscape** [comment]
Heh, true. Overall steem has a consistent block load and the "target fullness" of 25% should give us plenty of room to absorb any peaks. If we fix bandwidth only (a) and (c) make sense.

**Comment by TimCliff** [comment]
Ref https://github.com/steemit/condenser/issues/2298 for the FAQ reference part.

**Comment by imwatsi** [comment]
I've written another version of text we could use. Perhaps we could merge the two and come up with something.

---

# How does bandwidth work?

Bandwidth represents a share in the resources of the Steem blockchain's network and it is what enables users to perform actions like posting, voting, commenting and making token transfers. Every action a user takes on Steemit, with the exception of viewing content, consumes part of his/her account's bandwidth allocation. All accounts are allocated bandwidth according to how much Steem Power the account holds. The more the Steem Power, the more bandwidth is allocated.

This bandwidth rate-limiting is employed to safeguard the blockchain from spam attacks.

To see how much bandwidth your account has at any time, visit [https://steemd.com/](https://steemd.com)@your_username and look for the section shown in the screenshot below.

![Screenshot (3).png](https://steemitimages.com/DQmYwLpTW1RVY1jUKwLwEW9QSbEnP9DRW9h2BCerk3PfEDj/Screenshot%20(3).png)

Your account's bandwidth is measured in bytes, kilobytes, megabytes or gigabytes, and it recharges gradually over time. The progress bar labelled "bandwidth remaining" shows how much bandwidth you have left as a percentage.

## What happens when I run out of bandwidth?

Normally, most active accounts will have enough bandwidth to perform the usual actions on Steemit, but when there are a lot of people using the platform at a certain time, bandwidth allocations decrease as the bandwidth is spread amongst all active accounts. This leads to a decrease in the amount of bandwidth available. Accounts with very low amounts of Steem Power are more susceptible to bandwidth problems.

If your account runs out of bandwidth, you will not be able to perform actions on Steemit. To increase your bandwidth allocation, you can:

- Increase your Steem Power **(best option)**
- Wait until your bandwidth recharges, or
- Wait until the network's usage decreases

*To increase your Steem Power, you can buy more STEEM and power up, receive STEEM or Steem Power from someone else or have someone delegate some Steem Power to your account.*

**Comment by sneak** [comment]
Spec addition:  This should only display if the user *attempts to publish a transaction*, or *on the page with the editor, so they can know before they write a whole post*, and it should only display if their available bandwidth is zero.

**Comment by roadscape** [comment]
It seems if they're already at 0 or below we should show the warning unconditionally, since it affects the most common/basic op (votes).

**Comment by sneak** [comment]
No, because a user who isn't going to vote doesn't need to be warned about a not-uplifting condition *that doesn't affect them*.  It reduces UX for no benefit.  Remember, most users only ever lurk.

**Comment by roadscape** [comment]
Lurkers do not use bandwidth and thus will not be affected
